code/control/DeployDispatcher.php

```diff
@@ -76,31 +77,54 @@ public function history(SS_HTTPRequest $request) {
 		$start = ($page - 1) * $list->getPageLength();
 		$list->setPageStart((int) $start);
 		if (empty($list)) {
-			return $this->getAPIResponse(['message' => 'No deploy history'], 404);
+			return $this->getAPIResponse(['list' => []], 200);
 		}
 
+		$currentBuild = $this->environment->CurrentBuild();
+
 		foreach ($list as $deployment) {
 			$data[] = [
+				'ID' => $deployment->ID,
 				'CreatedDate' => $deployment->Created,
```

Comment:
I don't want to make a huge thing out of it, but I've never seen successful API that changes it output format . 🐹